packages/components/src/Disclosure/Disclosure.test.tsx

```diff
@@ -32,3 +33,35 @@ it("renders a Disclosure that is opened if `defaultOpen` is set", () => {
   const detailsElement = queryByRole("group");
   expect(detailsElement.getAttribute("open")).not.toBeNull();
 });
+
+describe("When a custom title is provided", () => {
+  it("renders the custom title", () => {
+    render(
+      <Disclosure
+        title={
+          <>
+            <Icon name="archive" />
+            <span>Custom Title</span>
+          </>
+        }
+      >
+        <span>Content</span>
+      </Disclosure>,
+    );
+
+    expect(screen.getByText("Custom Title")).toBeInTheDocument();
+    expect(screen.getByTestId("archive")).toBeInTheDocument();
+  });
+
+  it("renders the custom title with a custom class", () => {
+    render(
+      <Disclosure title={<span>Custom Title</span>}>
+        <span>Content</span>
+      </Disclosure>,
+    );
+
+    expect(screen.getByText("Custom Title").parentElement).toHaveClass(
+      "customSummaryWrap",
+    );
+  });
```

Comment:
I think in this case it would be more pragmatic to test the opposite/negative scenario, i.e. in case the string title is passed, the "customSummaryWrap" should not be on the parent:

```suggestion
  it("should not render a custom title if the passed 'title' is a string", () => {
    const title = "String Title";
    const customTitleClass = "customSummaryWrap";
    render(
      <Disclosure title={title}>
        <span>Content</span>
      </Disclosure>,
    );

    expect(screen.getByText(title).parentElement).not.toHaveClass(customTitleClass);
  });
```

Reply:
Yeah, you're right, I think it is better, 1sec.